packages/renderers-rust/README.md

```diff
@@ -37,12 +37,112 @@ codama.accept(renderVisitor(pathToGeneratedFolder, options));
+
+### Default traits
+
+Using the `traitOptions` attribute, you may configure the default traits that will be applied to every Rust type. These default traits can be configured using 4 different attributes:
+
+-   `baseDefaults`: The default traits to implement for all types.
+-   `dataEnumDefaults`: The default traits to implement for all data enum types, in addition to the `baseDefaults` traits. Data enums are enums with at least one non-unit variant — e.g. `pub enum Command { Write(String), Quit }`.
+-   `scalarEnumDefaults`: The default traits to implement for all scalar enum types, in addition to the `baseDefaults` traits. Scalar enums are enums with unit variants only — e.g. `pub enum Feedback { Good, Bad }`.
+-   `structDefaults`: The default traits to implement for all struct types, in addition to the `baseDefaults` traits.
+
+Note that you must provide the fully qualified name of the traits you provide (e.g. `serde::Serialize`). Here are the default values for these attributes:
+
+```ts
+const traitOptions = {
+    baseDefaults: [
+        'borsh::BorshSerialize',
+        'borsh::BorshDeserialize',
+        'serde::Serialize',
+        'serde::Deserialize',
+        'Clone',
+        'Debug',
+        'Eq',
+        'PartialEq',
+    ],
+    dataEnumDefaults: [],
+    scalarEnumDefaults: ['Copy', 'PartialOrd', 'Hash', 'num_derive::FromPrimitive'],
+    structDefaults: [],
+};
+```
+
+### Overridden traits
+
+In addition to configure the default traits, you may also override the traits for specific types. This will completely replace the default traits for the given type. To do so, you may use the `overrides` attribute of the `traitOptions` object.
+
+This attribute is a map where the keys are the names of the types you want to override, and the values are the traits you want to apply to these types. Here is an example:
+
+```ts
+const traitOptions = {
+    overrides: {
+        myCustomType: ['Clone', 'my::custom::Trait', 'my::custom::OtherTrait'],
+        myTypeWithNoTraits: [],
+    },
+};
+```
+
+### Feature Flags
+
+You may also configure which traits should be rendered under a feature flag by using the `featureFlags` attribute. This attribute is a map where the keys are feature flag names and the values are the traits that should be rendered under that feature flag. Here is an example:
+
+```ts
+const traitOptions = {
+    featureFlags: { fruits: ['fruits::Apple', 'fruits::Banana'] },
+};
+```
+
+Now, if at any point, we encounter a `fruits::Apple` or `fruits::Banana` trait to be rendered (either as default traits or as overridden traits), they will be rendered under the `fruits` feature flag. For instance:
+
+```rust
+#[cfg(feature = "fruits", derive(fruits::Apple, fruits::Banana))]
+```
+
+By default, the `featureFlags` option is set to the following:
+
+```ts
+const traitOptions = {
+    featureFlags: { serde: ['serde::Serialize', 'serde::Deserialize'] },
+};
+```
+
```

Comment:
I think we could also add a note to say that feature flags have to be present on the `Cargo.toml` as well.